domains/coupon-domain.js

```diff
@@ -11,12 +11,21 @@ function isCouponValid (coupon) {
   return !!coupon && validCouponType(coupon) && !!coupon.expired_at
 }
 
+function isCouponItemValid (coupon, item) {
```

Comment:
It would be hard for devs to understand difference between `isCouponItemValid` and `isCouponValid`. What difference would it be? When shall I use one of those?

It would be much easier with name like `canApply`, and we can also merge that into `isCouponValid` and let service use that instead.